### ISSUE TITLE
Equality on parameter string array of record 

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#9281]
### Description

[ClaRa.Components.Sensors.Check.TestGasSensors](https://libraries.openmodelica.org/branches/master/ClaRa_dev/files/ClaRa_dev_ClaRa.Components.Sensors.Check.TestGasSensors.sim) has connectors with media records that have an array of gas names (strings). For connections all parameters have to be equal. Similar to #9024 this model fails this check.

### Steps to Reproduce

Run

```modelica
loadFile("/path/to/TILMediaClaRa/TILMedia/package.mo"); getErrorString();
loadFile("/path/to/clara_om/ClaRa/package.mo"); getErrorString();
simulate(ClaRa_dev_ClaRa.Components.Sensors.Check.TestGasSensors); getErrorString();
```

```
assert            | warning | The following assertion has been violated during initialization at time 0.000000
|                 | |       | flueGasCell.outlet.Medium.gasNames[1] == gasFlowSource_T.gas_a.Medium.gasNames[1]
assert            | error   | Connected constants/parameters must be equal
```

### Version and OS

 - OpenModelica Version: v1.20.0-dev-240-g0763971d31
 - OS: Ubuntu Focal
 - ClaRa:  1.8.0

**Comment by AnHeuermann** [comment]
It seems that the array of strings have type `T_BOOL` instead of `T_STRING` inside the array. So something in SimCode or the Backend is going wrong.

**Comment by AnHeuermann** [comment]
@perost I'm not sure how to understand the `RELATION` generated in the frontend.
I added a `Operator.toString` (see https://github.com/OpenModelica/OpenModelica/pull/9282) and call it in [NFExpression.toDAE](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/Compiler/NFFrontEnd/NFExpression.mo#L2261):

```modelica
pressure.port.Medium.gasNames[10] == temperature.outlet.Medium.gasNames[10]
OPERATOR(Boolean[10], EQUAL)
```

where I would expect to get `OPERATOR(String[10], EQUAL)`. For scalar string equality this is indeed the case, e.g. for
```
mass_flow.outlet.Medium.concatGasName == temperature.inlet.Medium.concatGasName
OPERATOR(String, EQUAL)
```

**Comment by perost** [comment]
> @perost I'm not sure how to understand the `RELATION` generated in the frontend. I added a `Operator.toString` (see #9282) and call it in [NFExpression.toDAE](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/Compiler/NFFrontEnd/NFExpression.mo#L2261):
> 
> ```modelica
> pressure.port.Medium.gasNames[10] == temperature.outlet.Medium.gasNames[10]
> OPERATOR(Boolean[10], EQUAL)
> ```
> 
> where I would expect to get `OPERATOR(String[10], EQUAL)`. For scalar string equality this is indeed the case, e.g. for
> 
> ```
> mass_flow.outlet.Medium.concatGasName == temperature.inlet.Medium.concatGasName
> OPERATOR(String, EQUAL)
> ```

This might be fixed in #9292. For some reason we used `ComponentRef.getComponentType` to get the type to use when generating asserts for connect equations which doesn't take subscripts into account, so we'd sometimes get array types even though the argument were actually scalars. I'm not sure where the `Boolean` came from though...

Now we just take the type from the expressions we generate anyway, so they should be correct. The model now segfaults for me when calling some external function, which I guess might be progress?

**Comment by AnHeuermann** [comment]
> The model now segfaults for me when calling some external function, which I guess might be progress?

The generated code looks correct and for me it is simulating.  We'll see what the library tests will make from this, but this seems to be resolved. Thanks!
